### PR TITLE
Catch invalid range exception

### PR DESCRIPTION
**Pull request by jgoizueta**
See #397

### CODE REVIEW

File: test/integration/factory.js [hunk 1]
```diff
@@ -51,4 +56,56 @@ describe('factory', function() {
...
         });
     });
+
+    it('invalid filters in createCacheTable', function(done) {
+        const TEST_SOURCE_TYPE = 'test-source';
+        const TestSource = Node.create(TEST_SOURCE_TYPE, {
+            table: Node.PARAM.STRING()
+        }, { cache: true });
+        TestSource.prototype.sql = function() {
+            return 'select * from ' + this.table;
```
Comment (dgaubert) [inline test/integration/factory.js]:
```suggestion
            return `select * from ${this.table}`;
```

File: test/integration/factory.js [hunk 2]
```diff
@@ -51,4 +56,56 @@ describe('factory', function() {
...
+                source: sourceDefinition,
+                sampling: 1,
+                seed: 12345,
+                filters: invalid_filters
+            }
+        };
+        const definition = {
+            type: 'sampling',
+            params: {
+                source: samplingDefinition,
```
Comment (dgaubert) [inline test/integration/factory.js]: Do we need to nest two sampling nodes?

File: test/integration/factory.js [hunk 3]
```diff
@@ -51,4 +56,56 @@ describe('factory', function() {
             return done();
         });
     });
+
+    it('invalid filters in createCacheTable', function(done) {
+        const TEST_SOURCE_TYPE = 'test-source';
+        const TestSource = Node.create(TEST_SOURCE_TYPE, {
+            table: Node.PARAM.STRING()
+        }, { cache: true });
+        TestSource.prototype.sql = function() {
```
Comment (dgaubert) [inline test/integration/factory.js]: Long time I worked on this library, so I might forget a lot of context. So my following question might not make sense: why do we need a `test-source` type?

Comment (jgoizueta) [reply test/integration/factory.js]: Good question, I'm not sure either, I picked it from the existing test in that file, it's a cached source, it worked for me so I used it (but I guess I could use an actual Source).

File: test/integration/factory.js (hunk 2 above)
Comment (jgoizueta) [reply test/integration/factory.js]: Yes, I could only reproduce the exception within `createTableQuery` that way. With just one the exception occurs after createTableQuery (in getColumns) where it is already handled correctly.

File: test/integration/factory.js [hunk 4]
```diff
@@ -51,4 +57,56 @@ describe('factory', function() {
...
+            return 'select * from ' + this.table;
+        };
+
+        const invalid_filters = {
```
[the remark this answers is not in the document]
Comment (jgoizueta) [reply test/integration/factory.js]: oops

File: lib/service/database.js [hunk 5]
```diff
@@ -110,14 +110,18 @@ DatabaseService.prototype.createCacheTable = function(node, callback) {
...
-            self.queryRunner.run(sqlWrappedNode.createTableQuery(), QUERY_RUNNER_WRITE_OP, function(err) {
-                debug('TODO: register table');
-                if (err && !DUPLICATED_TABLE_ERROR_CODES[err.code]) {
-                    return callback(err, tableExists);
-                }
-
-                return callback(null, tableExists);
-            });
+            try {
+                self.queryRunner.run(sqlWrappedNode.createTableQuery(), QUERY_RUNNER_WRITE_OP, function(err) {
```
[the remark this answers is not in the document]
Comment (jgoizueta) [reply lib/service/database.js]: you're right there, we need to catch just around createTableQuery

**Pull request merged**